localstack/config.py

```diff
@@ -867,11 +867,17 @@ def __iter__(self):
 
 SERVICE_PROVIDER_CONFIG = ServiceProviderConfig("default")
 
-for key, value in os.environ.items():
-    if key.startswith("PROVIDER_OVERRIDE_"):
-        SERVICE_PROVIDER_CONFIG.set_provider(
-            key.lstrip("PROVIDER_OVERRIDE_").lower().replace("_", "-"), value
-        )
+
+def process_provider_overrides(provider_config):
+    override_prefix = "PROVIDER_OVERRIDE_"
+    for key, value in os.environ.items():
+        if key.startswith(override_prefix):
+            provider_config.set_provider(
+                key[len(override_prefix) :].lower().replace("_", "-"), value
+            )
```

Comment:
could we maybe make this function part of the `ServiceProviderConfig` and call it `load_from_environment(env=os.environ)`?

Reply:
Excellent point. Should be better now :)